CHANGELOG.md

```diff
@@ -1,5 +1,274 @@
+common double dash `--` option break indicator:
+
+```
+$ kontena master ssh -- ls -al
+ERROR: Unrecognised option '-l'
+$ kontena master ssh -- ls -al
+$ kontena vault write -- SECRET --secret-password--
+```
+
+#### Kontena Stack Registry V2 API And The New 'meta' Fields
+
+While mostly invisible to the end-user, the CLI stack registry API client is now using
+the completely rewritten stack registry and the V2 JSON-API it offers. The registry
+supports GZip responses, private stacks, server-side stack YAML validation and parsing
+of the new top level 'meta:' fields.
+
+The meta fields can be used to add extra information to stacks published in the registry.
+
+You can find the full set of accepted metadata fields in the pull request #3219 description.
+
+As the CLI HTTP client now supports gzip compressed responses, we have also added the option
+to enable compression in the Kontena Master API. To enable, set `KONTENA_SERVER_GZIP=true`
+in the Master environment.
+
+#### Drop Support For Ruby 2.1, Build Installer With Embedded Ruby 2.5.0
+
+As Ruby 2.1 branch has been out of development for almost a year now, it's time to upgrade
+if you already didn't.
+
+The MacOS Kontena CLI installation package is now bundled with Ruby version 2.5.0
+
+Ruby 2.2 is nearing its EOL at the end of March 2018.
+
+#### Process Multiple Items In One Command
+
+Many of the subcommands can now accept a list of items instead of just one. This is handy in
+shell scripts and one-liners, for example:
+
+```
+$ kontena vault ls -q | xargs kontena vault rm --force
+$ kontena vault rm --force $(kontena vault ls -q)
+```
+
+#### Master Authentication Token Descriptions
+
+You can now add descriptions to the master authentication tokens:
+
+```
+$ kontena master token create -e 0 --description "deploy key"
+$ kontena master token ls
+ID                         TOKEN_TYPE   TOKEN_LAST4   EXPIRES_IN   SCOPES       DESCRIPTION
+5a8c275351d1a1001566a4ef   bearer       f539          never        user         deploy key
+```
+
+#### Health Check
+
+The agent now uses the port in health check definition when configuring the load balancer. (PR [#3113](https://github.com/kontena/kontena/pull/) / Issue [#1709](https://github.com/kontena/kontena/issues/1709))
+
+Example configuration: 
+
+```
+    health_check:
+      protocol: http
+      uri: /
+      port: 8000
+```
+
+The health check will now consider HTTP 3XX status codes as healthy. (PR [#3265](https://github.com/kontena/kontena/pull/3265) / Issue [#1790](https://github.com/kontena/kontena/issues/1790))
+
+#### Logging Container Crashes
+
+It was previously not possible to see if a container restarted because it crashed or if it was intentional and caused by for example a deploy or a manual restart. ([#3286](https://github.com/kontena/kontena/pull/3286))
+
+```
+2018-02-16T14:43:26.731698302Z container die 9d21e309419ffbd32d75ab4bf544baf4deefb491934a762fc88b5c34a3071a52 (exitCode=137...)
```

Comment:
This example is bad, it's from the `docker events` output. It should be from the `kontena service events` output instead:

```
TIME                      TYPE                 MESSAGE
...
2018-02-16T12:38:38.583Z  instance_crash       service test/client-1 instance exited with code 1, restarting (delay: 0s) (core-01)
2018-02-16T12:38:38.951Z  start_instance       starting service instance /test/client-1 (core-01)
2018-02-16T12:38:39.513Z  start_instance       service instance /test/client-1 started successfully (core-01)
```